x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichResiliencyTests.java

```diff
@@ -0,0 +1,249 @@
+            enrichPolicyName,
+            new EnrichPolicy(EnrichPolicy.MATCH_TYPE, null, List.of(enrichIndexName), "my_key", List.of("my_value"))
+        )).actionGet();
+
+        client().execute(ExecuteEnrichPolicyAction.INSTANCE, new ExecuteEnrichPolicyAction.Request(enrichPolicyName)
+            .setWaitForCompletion(true)).actionGet();
+
+        XContentBuilder pipe1 = JsonXContent.contentBuilder();
+        pipe1.startObject();
+        {
+            pipe1.startArray("processors");
+            {
+                pipe1.startObject();
+                {
+                    pipe1.startObject("enrich");
+                    {
+                        pipe1.field("policy_name", enrichPolicyName);
+                        pipe1.field("field", "custom_id");
+                        pipe1.field("target_field", "enrich_value_1");
+                    }
+                    pipe1.endObject();
+                }
+                pipe1.endObject();
+                pipe1.startObject();
+                {
+                    pipe1.startObject("enrich");
+                    {
+                        pipe1.field("policy_name", enrichPolicyName);
+                        pipe1.field("field", "custom_id");
+                        pipe1.field("target_field", "enrich_value_2");
+                    }
+                    pipe1.endObject();
+                }
+                pipe1.endObject();
+            }
+            pipe1.endArray();
+        }
+        pipe1.endObject();
+
+        client().execute(PutPipelineAction.INSTANCE, new PutPipelineRequest(
+            enrichPipelineName,
+            BytesReference.bytes(pipe1),
+            XContentType.JSON
+        )).actionGet();
+
+        client().admin().indices().create(new CreateIndexRequest(enrichedIndexName)).actionGet();
+
+        XContentBuilder doc = JsonXContent.contentBuilder().startObject().field("custom_id", "key").endObject();
+
+        BulkRequest bulk = new BulkRequest(enrichedIndexName);
+        bulk.timeout(new TimeValue(10, TimeUnit.SECONDS));
+        for (int idx = 0; idx < 50; idx++) {
+            bulk.add(new IndexRequest().source(doc).setPipeline(enrichPipelineName));
+        }
+
+        BulkResponse bulkItemResponses = client().bulk(bulk).actionGet(new TimeValue(30, TimeUnit.SECONDS));
+
+        assertTrue(bulkItemResponses.hasFailures());
+        for (BulkItemResponse item : bulkItemResponses.getItems()) {
+            if (item.isFailed()) {
+                assertThat(item.getFailure().getStatus().getStatus(), is(equalTo(429)));
+                assertThat(item.getFailureMessage(), containsString("Could not perform enrichment, enrich coordination queue at capacity"));
+                break;
+            }
+        }
+
+        client().admin().indices().refresh(new RefreshRequest(enrichedIndexName)).actionGet();
+        logger.info(client().search(new SearchRequest(enrichedIndexName)).actionGet().toString());
```

Comment:
perhaps assert that the docs that have been successfully enriched (having both `enrich_value_1` and `enrich_value_2` fields) is equal to the number of successful bulk items?